File: lang/ruby/src/muse/algorithms/sorting/SelectionSort.rb
module SelectionSort

    def self.sort(arr)
        if arr.empty?
            return
        end

        iMin = 0

        bound = arr.length - 1
        0.upto(bound - 1) do |i|
            iMin = i
            (i + 1).upto(bound) do |j|
                if arr[j] < arr[iMin]
                    iMin = j
                end
            end
            if iMin != i
                arr[i], arr[iMin] = arr[iMin], arr[i]
            end
        end
    end
end

File: go/muse/algorithms/sorting/SelectionSort/SelectionSort.go
package SelectionSort

func Sort(arr []int) {
	if len(arr) == 0 {
		return
	}

	var iMin int

	for i, bound := 0, len(arr)-1; i < bound; i++ {
		iMin = i
		for j := i + 1; j <= bound; j++ {
			if arr[j] < arr[iMin] {
				iMin = j
			}
		}
		if iMin != i {
			arr[i], arr[iMin] = arr[iMin], arr[i]
		}
	}
}
